finagle-core/src/test/scala/com/twitter/finagle/service/StatsFilterTest.scala

```diff
@@ -141,6 +144,21 @@ class StatsFilterTest extends FunSuite {
     assert(receiver.gauges(Seq("pending"))() == 0.0)
   }
 
+  test("don't report pending requests after uncaught exceptions") {
+    val receiver = new InMemoryStatsReceiver()
+    val service = new Service[String, String] {
+      def apply(request: String): Future[String] = throw new Exception("broken")
+    }
+    val statsFilter = new StatsFilter[String, String](receiver, BasicExceptions)
+    val chain = new Service[String, String] {
```

Comment:
This is on purpose. If I use `andThen` then this is already [wrapping](https://github.com/twitter/finagle/blob/develop/finagle-core/src/main/scala/com/twitter/finagle/Filter.scala#L77) `NonFatal`'s. Which means that the test wouldn't trigger the problem we are trying to fix.